### DISCUSSION TITLE
Pydantic v2 support

### DISCUSSION THREAD

**Discussion by frozenBranch** [fastapi/sqlmodel#621] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
pip install fastapi==0.100.0 pydantic==2.0.3

## sqlmodel 0.0.8 requires pydantic<2.0.0,>=1.8.2, but you have pydantic 2.0.3 which is incompatible.
```

### Description

Hi, big fan of fastapi, sqlmodel and pydantic here!
I'm currently using this stack in 2 projects, I was trying to update fastapi and pydantic tro try the new pydantic v2, but I'm currently getting this error
`sqlmodel 0.0.8 requires pydantic<2.0.0,>=1.8.2,`
Was wondering if there is currently a workaround to be able to use pydantic v2, or if there a is a plan to be compatible with pydantic v2.
Not demanding a fixed timeline, I know OSS can be harsh so just wanted to know if it was on sqlmodel's plans to support the pydantic v2 package in the foreseeable future.

### Operating System

Linux, macOS

### SQLModel Version

0.0.8

### Python Version

3.11.3

**Comment by grski** [comment]
Second this, I'm facing a similar problem that's blocking the usage of the package or update of the fastapi :( I'm happy to help if I can.

**Comment by antont** [comment]
Tiangolo has upgraded FastAPI to Pydantic v2, and said last week that he'll port SQLModel next. Let's see when he gets to it, hopefully soon. If not, there's a nice migration guide from Pydantic folks that maybe makes it easy enough for someone else too.

**Comment by grski** [comment]
I had a few minutes to spare and started the work, slowly: https://github.com/tiangolo/sqlmodel/pull/622

**Comment by DarylStark** [comment]
Also important for me. Pretty sure Sebastian is already working on it :)

**Reply by cdrcqnts** [reply]
Still get the error: `sqlmodel 0.0.8 requires pydantic<2.0.0,>=1.8.2, but you have pydantic 2.3.0 which is incompatible.` Any update on this?

**Comment by heyaco** [comment]
Should I fork to add pydantic/sqlalchemy 2 support or wait? If it's in the works I'd like to contribute.

**Reply by antont** [reply]
people seem to be using the fork at https://github.com/honglei/sqlmodel

it's discussed in https://github.com/tiangolo/sqlmodel/pull/632

**Comment by mrdeveloperdude** [comment]
FastAPI supports Pydantic >=2.0.0 since version 0.100.0 but that does not help when all the packages you end up depending on in your FastAPI application only support <2.0.0. pip can't resolve dependencies and my app is a paperweight without any way forward. I need Pydantic >=2.0.0 due to an ugly bug in the older versions.

**Comment by YuriiMotov** [comment]
SQLModel now supports Pydantic V2
